File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        final var indexesByNums = new HashMap<Integer, Integer>();
        
        for (int i = 0; i < nums.length; i++) {
            
            final var currentValue = nums[i];
            final var additionalValue = target - currentValue;
            final var additionalValueIndex = indexesByNums.get(additionalValue);
            
            if (additionalValueIndex != null) {
                return new int[] { i, additionalValueIndex };
            }
            
            indexesByNums.put(currentValue, i);
        }
        
        return new int[0];
    }
}

File: 1-two-sum/two-sum.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        
        var indexesByNums = Dictionary<Int, Int>()

        for i in 0..<nums.count {
            
            let currentValue = nums[i]
            let complimentaryNumber = target - currentValue

            if let indexOfComplimentaryNumber = indexesByNums[complimentaryNumber] {
                return [i, indexOfComplimentaryNumber]
            }

            indexesByNums[currentValue] = i
        }

        return []
    }
}